GitCommands/Settings/FileSettingsCache.cs

```diff
@@ -163,6 +157,13 @@ protected override void SaveImpl()
                     }
                 }
 
+                // ensure the directory structure exists
```

Comment:
This creation (as well as the Copy below) should be wrapped in a try-catch too, to handle when the settings path cannot be written to

Reply:
What would we do if we can't write to the path apart from a crash?

Reply:
We could just ignore or show an popup (that should be limited to one per GE startup).
There are other issues related to this.
But that could be changed separately
Will approve later today, unless someone has any comments as this will be merged at my approval

Reply:
> We could just ignore or show an popup (that should be limited to one per GE startup).

NBug will intercept this and presumably show the error to a user and, likely, to us. I think this is an acceptable solution, we have bigger fish to fry atm.

Reply:
> NBug will intercept this and presumably show the error to a user and, likely, to us. I think this is an acceptable solution, we have bigger fish to fry atm.

Neither of these are big issues. But I consider the problem to write settings is worse than this. But we take that separately.